resident/resident-service/src/main/java/io/mosip/resident/util/Utilitiy.java

```diff
@@ -318,4 +321,14 @@ public ResidentTransactionEntity createEntity() throws ApisResourceAccessExcepti
 		return residentTransactionEntity;
 	}
 
+	public static boolean isSecureSession(){
+		if(SecurityContextHolder.getContext()!=null){

```

Comment:
replace the statement with:
`return Optional.ofNullable(SecurityContextHolder.getContext())
			.map(SecurityContext::getAuthentication)
			.map(Authentication::getPrincipal)
			.map(obj -> obj.equals(ANONYMOUS_USER))
			.isPresent();`

Reply:
Done.